### ISSUE TITLE
S1125: Consider throw expression case

### ISSUE THREAD

**Issue by anreton** [SonarSource/sonar-dotnet#4528]
### Description
[S1125](https://rules.sonarsource.com/csharp/RSPEC-1125)

### Repro steps

```csharp
(key is null) ? throw new ArgumentNullException(nameof(key)) : false;
```

### Expected behavior

No false diagnostic for S1125.

### Actual behavior

Get false diagnostic for S1125.

And codefix is incorrect too.
![image](https://user-images.githubusercontent.com/28877846/121266425-26cbcb00-c8c3-11eb-8ba7-4e9330d48911.png)

### Related information

- Microsoft Visual Studio Community 2019 Version 16.10.0
- SonarAnalyzer.CSharp 8.23.0.32424 via Nuget
- https://stackoverflow.com/questions/46282957/throw-expressions-not-working-for-boolean-expressions
- https://stackoverflow.com/questions/56613707/is-there-a-true-or-throw-or-non-null-or-throw-expression-in-c

**Comment by pavel-mikula-sonarsource** [comment]
Hi @anreton ,

Thank you for reporting this case, I can confirm it as FP. We already have an issue for it that you can track #4465 so I'll close this one.
